sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/aggregator/DoubleExponentialHistogramBuckets.java

```diff
@@ -234,19 +257,44 @@ public boolean equals(@Nullable Object obj) {
     DoubleExponentialHistogramBuckets other = (DoubleExponentialHistogramBuckets) obj;
     // Don't need to compare getTotalCount() because equivalent bucket counts
     // imply equivalent overall count.
-    return getBucketCounts().equals(other.getBucketCounts())
-        && this.getOffset() == other.getOffset()
-        && this.scale == other.scale;
+    return this.scale == other.scale && sameBucketCounts(other);
+  }
+
+  /**
+   * Tests if two bucket counts are equivalent semantically.
+   *
+   * <p>Semantic equivalence means:
+   *
+   * <ul>
+   *   <li>All counts are stored between indexStart/indexEnd.
+   *   <li>Offset does NOT need to be the same
+   * </ul>
+   */
+  private boolean sameBucketCounts(DoubleExponentialHistogramBuckets other) {
+    int min = Math.min(this.counts.getIndexStart(), other.counts.getIndexStart());
```

Comment:
Can you confirm this logic? Shouldn't it be something like `a = this.start, b = other.start, a <= this.end && b <= other.end; a++ && b++)`?

It looks like we could refer to an index that is below `start` for one of the two buckets right now

Reply:
Yes, actually that's the point.

What can happen right now (with `diff`) is that we have a bucket, e.g. at index 2 which "diffs" to 0.   In the unit tests, we instantiate the histogram-buckets by saying "You should see count 5 at index 3", which means we have one bucket with index start @ 3 and another with index start @ 2.   They're semantically equivalent counts.

The previous MapCounter implementation actually did GC so any count that reached zero was removed from the Map.  Unfortunately, that's not really feasible in the circular buffer implementation.

Reply:
This is what you are trying to say, right? 

```
bucket counts: 
0,0,5,1,2,0,0, == 5,1,2
```

Makes sense to me.

Reply:
added